app/javascript/mastodon/features/ui/hooks/useBreakpoint.tsx

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+
+const breakpoints = {
+  openable: 759,
+  full: 1174,
+};
+
+type Breakpoint = 'openable' | 'full';
+
+export const useBreakpoint = (breakpoint: Breakpoint) => {
+  const [isMatching, setIsMatching] = useState(false);
+
+  useEffect(() => {
+    const mediaWatcher = window.matchMedia(
+      `(max-width: ${breakpoints[breakpoint]}px)`,
+    );
+
+    setIsMatching(mediaWatcher.matches);
+
+    const handleChange = (e: MediaQueryListEvent) => {
+      setIsMatching(e.matches);
+    };
+
+    mediaWatcher.addEventListener('change', handleChange);
+
+    return () => {
+      mediaWatcher.removeEventListener('change', handleChange);
+    };
+  }, [breakpoint, setIsMatching]);
+
+  return isMatching;
+};
+
+interface WithBreakpointType {
+  matchesBreakpoint: boolean;
+}
+
+export function withBreakpoint<P>(
+  Component: React.ComponentType<P & WithBreakpointType>,
+) {
+  const displayName = `withMobileLayout(${Component.displayName ?? Component.name})`;
+
+  const ComponentWithBreakpoint = (props: P) => {
+    const matchesBreakpoint = useBreakpoint('full');
```

Comment:
It's not obvious that `withBreakpoint` only supports a single fixed breakpoint, but it's probably ok for now and could be extended later if necessary.

Reply:
Added a way to pass the value.